pkg/ebpf/tracer.go

```diff
@@ -237,23 +229,72 @@ func NewTracer(config *Config) (*Tracer, error) {
 		destExcludes:   network.ParseConnectionFilters(config.ExcludedDestinationConnections),
 	}
 
-	tcpCloseMap, _ := tr.getMap(tcpCloseBatchMap)
-	batchManager, err := NewPerfBatchManager(m, tcpCloseMap, config.TCPClosedTimeout)
+	tcpCloseMap, _ := tr.getMap(bytecode.TcpCloseBatchMap)
+	batchManager, err := NewPerfBatchManager(tcpCloseMap, config.TCPClosedTimeout)
 	if err != nil {
 		return nil, err
 	}
 	tr.batchManager = batchManager
 
-	tr.perfMap, err = tr.initPerfPolling()
+	tr.perfMap, err = tr.initPerfPolling(perfHandler)
 	if err != nil {
 		return nil, fmt.Errorf("could not start polling bpf events: %s", err)
 	}
 
+	if err = m.Start(); err != nil {
+		return nil, fmt.Errorf("could not start ebpf manager: %s", err)
+	}
+
 	go tr.expvarStats()
 
 	return tr, nil
 }
 
+func overrideProbeSectionNames(m *manager.Manager) {
+	for _, p := range m.Probes {
+		if !p.Enabled {
+			continue
+		}
+		if override, ok := bytecode.KProbeOverrides[bytecode.ProbeName(p.Section)]; ok {
+			p.Section = string(override)
+		}
+	}
+}
+
+func runOffsetGuessing(config *Config, buf *bytes.Reader) ([]manager.ConstantEditor, error) {
+	// Enable kernel probes used for offset guessing.
+	offsetMgr := bytecode.NewOffsetManager()
+	offsetOptions := manager.Options{
+		RLimit: &unix.Rlimit{
+			Cur: math.MaxUint64,
+			Max: math.MaxUint64,
+		},
+	}
+	for _, probeName := range offsetGuessProbes(config) {
+		offsetOptions.ActivatedProbes = append(offsetOptions.ActivatedProbes, string(probeName))
+	}
+	if err := offsetMgr.InitWithOptions(buf, offsetOptions); err != nil {
+		return nil, fmt.Errorf("could not load bpf module for offset guessing: %s", err)
+	}
+
+	if err := offsetMgr.Start(); err != nil {
+		return nil, fmt.Errorf("could not start offset ebpf manager: %s", err)
+	}
+	defer func() {
+		err := offsetMgr.Stop(manager.CleanAll)
```

Comment:
Glad to see we're finally disabling the Kprobes used exclusively for offset guessing.